HHS/Head-Start-TTADP
# [Prod] Filter similarity API results by region ID

## Description of change
There are some recipients that have grants across regions (I believe there are 3). The similarity API did not filter the results by region, but the rest of the app does, so those recipients would see "phantom" results. This PR fixes that

## How to test
- Visit https://tta-smarthub-staging.app.cloud.gov/recipient-tta-records/1411/region/5/rttapa. Note the number of similarity goals. Visit each goal; none should take you to blank pages
- Visit https://tta-smarthub-staging.app.cloud.gov/recipient-tta-records/1411/region/9/rttapa. Note the number of similarity goals. The groups should be different than the previous page. Visit each goal; none should take you to blank pages

## Issue(s)
* https://jira.acf.gov/browse/TTAHUB-2736

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
